File: leetcode/0268-missing-number/missingNumber.go
package leetcode

func missingNumber(nums []int) int {
	var result = 0
	var numMap = map[int]struct{}{}

	for _, num := range nums {
		numMap[num] = struct{}{}
	}

	for i := 0; i <= len(nums); i++ {
		if _, ok := numMap[i]; ok {
			continue
		}

		result = i
		break
	}

	return result
}

File: leetcode/0268-missing-number/missingNumber.ts
function missingNumber(nums: number[]): number {
  let result = 0;
  const numberSet = new Set<number>();
  for (const num of nums) {
    numberSet.add(num);
  }

  for (let index = 0; index <= nums.length; index++) {
    if (numberSet.has(index)) {
      continue;
    }

    result = index;
    break;
  }

  return result;
}

export default missingNumber;
